Fix common device computation for comparison ops.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66245

Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

Differential Revision: [D31503849](https://our.internmc.facebook.com/intern/diff/D31503849)

ysiraichi committed:
Fix common device computation for comparison ops.

Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

[ghstack-poisoned]

ysiraichi committed:
Update on "Fix common device computation for comparison ops."


Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

[ghstack-poisoned]

ezyang reviewed (approved):
aye aye matey

ysiraichi committed:
Update on "Fix common device computation for comparison ops."


Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

Differential Revision: [D31503849](https://our.internmc.facebook.com/intern/diff/D31503849)

[ghstack-poisoned]

ysiraichi committed:
Update on "Fix common device computation for comparison ops."


Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

Differential Revision: [D31503849](https://our.internmc.facebook.com/intern/diff/D31503849)

[ghstack-poisoned]

ysiraichi commented:
@ezyang @ngimel This is a friendly reminder. Do you have some time to take a look at this PR?

ngimel commented:
Can you please restore `.declare_static_dtype_and_device`? Torchvision depends on it.

ysiraichi committed:
Update on "Fix common device computation for comparison ops."


Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

Differential Revision: [D31503849](https://our.internmc.facebook.com/intern/diff/D31503849)

[ghstack-poisoned]

ysiraichi committed:
Update on "Fix common device computation for comparison ops."


Fixes #66053

This PR splits `declare_static_dtype_and_device` into two new methods for
`TensorIteratorBase`: `declare_static_dtype` and `declare_static_device`.

Differential Revision: [D31503849](https://our.internmc.facebook.com/intern/diff/D31503849)

[ghstack-poisoned]
